src/internal/components/chart-series-details/index.tsx

```diff
@@ -1,49 +1,117 @@
-import React, { memo } from 'react';
+import React, { memo, ReactNode, useEffect, useRef } from 'react';
 import clsx from 'clsx';
 
 import { BaseComponentProps, getBaseProps } from '../../base-component';
-import InternalBox from '../../../box/internal';
-import { ChartDetailPair } from '../../../pie-chart/interfaces';
 import ChartSeriesMarker, { ChartSeriesMarkerType } from '../chart-series-marker';
 import styles from './styles.css.js';
+import InternalExpandableSection from '../../../expandable-section/internal';
+import getSeriesDetailsText from './series-details-text';
+
+interface ChartDetailPair {
+  key: ReactNode;
+  value: ReactNode;
+}
 
 export interface ChartSeriesDetailItem extends ChartDetailPair {
   markerType?: ChartSeriesMarkerType;
   color?: string;
   isDimmed?: boolean;
+  subItems?: ReadonlyArray<ChartDetailPair>;
+  expandable?: boolean;
 }
+export type ExpandedStates = Record<number, boolean>;
 
 export interface ChartSeriesDetailsProps extends BaseComponentProps {
   details: ReadonlyArray<ChartSeriesDetailItem>;
+  expandedStates?: ExpandedStates;
+  setPopoverText?: (s: string) => void;
+  setExpandedState?: (index: number, state: boolean) => void;
 }
 
 export default memo(ChartSeriesDetails);
 
-function ChartSeriesDetails({ details, ...restProps }: ChartSeriesDetailsProps) {
+function ChartSeriesDetails({
+  details,
+  expandedStates,
+  setPopoverText,
+  setExpandedState,
+  ...restProps
+}: ChartSeriesDetailsProps) {
   const baseProps = getBaseProps(restProps);
   const className = clsx(baseProps.className, styles.root);
+  const detailsRef = useRef<HTMLDivElement | null>(null);
+
+  // Once the component has rendered, pass its content in plain text
+  // so that it can be used by screen readers.
+  useEffect(() => {
+    if (setPopoverText) {
+      if (detailsRef.current) {
+        setPopoverText(getSeriesDetailsText(detailsRef.current));
+      }
+      return () => {
+        setPopoverText('');
+      };
+    }
+  }, [details, setPopoverText]);
 
   return (
-    <div {...baseProps} className={className}>
+    <div {...baseProps} className={className} ref={detailsRef}>
       <ul className={styles.list}>
-        {details.map(({ key, value, markerType, color, isDimmed }, index) => (
+        {details.map(({ key, value, markerType, color, isDimmed, subItems, expandable }, index) => (
           <li
             key={index}
             className={clsx({
               [styles.dimmed]: isDimmed,
               [styles['list-item']]: true,
+              [styles['with-sub-items']]: subItems?.length,
+              [styles.expandable]: expandable,
             })}
           >
-            <div className={styles.key}>
-              {markerType && color && <ChartSeriesMarker type={markerType} color={color} />}
-              <span>{key}</span>
-            </div>
-            <InternalBox textAlign="right">{value}</InternalBox>
+            {subItems?.length && expandable ? (
```

Comment:
not: can this be simplified a bit with introducing more components e.g. <ExpandableItemsGroup subItems={subItems} ... />, <StaticItemsGroup subItems={subItems} ... />?

```
{hasSubItems && expandable ? <ExpandableitemsGroup ... /> : <StaticItemsGroup ... />}
```

Reply:
Refactored [here](https://github.com/cloudscape-design/components/pull/1644/commits/42cb7b1cb1f3834d85fbd5b0bad23f7cd9222143)